### ISSUE TITLE
Sync to Teams not working because of version format issues

### ISSUE THREAD

**Issue by oppie85** [SharePoint/sp-dev-docs#9261]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Describe the bug / error

We are running into a strange bug when trying to sync an SPFX solution to teams from the app catalog; it always fails with a 400 (bad request) response, even for packages that did sync successfully before.

I've managed to narrow down what the problem is; when I downloaded the app package using the "https://[yourtenant]].sharepoint.com/sites/appcatalog/_api/web/tenantappcatalog/downloadteamssolution(12)/$value" command and then tried to add it to Teams manually, it came up with the following error:

**App version supports only numbers in the '#.#.#' format.**

After manually changing the version number to comply with this structure, it could be added to Teams successfully. 

I then changed the version number of my spfx component to the same '#.#.#' format, but then it can't be added to the app catalog. 
So basically since SharePoint expects a '#.#.#.#' version and Teams expects a '#.#.#' version, it is currently impossible to use the 'Sync to Teams' feature at all. 

Is anyone else experiencing this behaviour?

### Steps to reproduce

1. Create an SPFx webpart solution, build it and add it to the app catalog
2. Deploy the SPFx webpart to all sites
3. Try to use the 'Sync to Teams' button

### Expected behavior

Expected behaviour would be for the 'Sync to Teams' step to create a valid Teams manifest or for Teams to accept '#.#.#.#' version numbers.

**Comment by nick-pape** [comment]
Hi there. Thanks for reporting this issue. We've confirmed this with a local repro and are investigating root cause currently.

**Comment by AJIXuMuK** [comment]
This has been fixed and will be rolled out soon.

**Comment by AJIXuMuK** [comment]
The fix should be rolled out.
